### ISSUE TITLE
[precise-gce] SGE slow to respond/run (degree of parallelism issue)

### ISSUE THREAD

**Issue by jakirkham** [travis-ci/travis-ci#5347]
Spun out from https://github.com/travis-ci/travis-ci/issues/5292

For example, this is an SGE test from before the change ( https://travis-ci.org/nanshe-org/nanshe/builds/83350103#L2553 ). The test to look for is called `tests.test_nanshe.test_learner.test_main_3`. The `tests.test_nanshe.test_learner.test_main_2` uses spawning (fresh interpreter) and `tests.test_nanshe.test_learner.test_main_1` is not parallel at all. Each of these works with the same parameters and data, but handle it differently. Here you can see this test takes ~20s.

On the recent working SGE build `tests.test_nanshe.test_learner.test_main_3` took ~190s. See the equivalent line here ( https://travis-ci.org/nanshe-org/nanshe/builds/97751627#L2620 ). Note that `tests.test_nanshe.test_learner.test_main_2` takes a similar amount of time as before.

I can understand SGE test lagging a bit, which does happen. However, this is a bit more than lag. Is it possible that there is some networking issue that is causing this?

**Comment by jakirkham** [comment]
@BanzaiMan, I remember the old VMs had 1.5 parallelism. What is the parallelism on GCE?

**Comment by jakirkham** [comment]
I have change the SGE tests that I had so that they should run much faster even if there is no real parallelism. So, will close this out.

The issue is we have less parallelism now than we did before so my queue size with SGE is smaller (as it is set based on the number of cores available). This could be overridden to make a large queue (larger than the number of cores) even if it won't allow true parallelism so we aren't waiting for each job individually to complete.

However, cutting down the number of jobs in the queue will also help, which is what I did instead.
